firebase-segmentation/src/test/java/com/google/firebase/segmentation/FirebaseSegmentationTest.java

```diff
@@ -184,7 +199,11 @@ public void testUpdateCustomInstallationId_CacheError_BackendOk() throws Interru
 
     // Expect exception
     try {
-      Tasks.await(firebaseSegmentation.setCustomInstallationId(CUSTOM_INSTALLATION_ID));
+      TestOnCompleteListener<Void> onCompleteListener = new TestOnCompleteListener<>();
+      firebaseSegmentation
+          .setCustomInstallationId(CUSTOM_INSTALLATION_ID)
+          .addOnCompleteListener(taskExecutor, onCompleteListener);
+      onCompleteListener.await();
```

Comment:
Optional: We can move this into an initialization method.

Reply:
Hmmm, first, I don't think we can put this entire block into initialization() since it's not always the same in every test case.

If you meant that move TestOnCompleteListener<Void> onCompleteListener = new TestOnCompleteListener<>() into initialization, I also think putting here is better since it's a whole logic block with the two lines below.  Also, if a test case uses  onCompleteListener more than once, then putting it in initialization is also not perfect.